ui/src/app/pages/analysis/util.ts

```diff
@@ -0,0 +1,15 @@
+export function dropNotebookFileSuffix(filename: string) {
```

Comment:
I like it!  Could you also add `const notebookExtension = '.ipynb'` for internal use here?